### ISSUE TITLE
Spacer guests

### ISSUE THREAD

**Issue by OrionFive** [OrionFive/Hospitality#288]
A more demanding faction of space faring guests who can show up.

**Comment by OrionFive** [comment]
Alright, here are some thoughts about the matter. 

- Spacers aren't an existing faction, so the old faction relationship stuff won't work. This will probably require some rewriting here and there.
- Creating a custom faction probably doesn't do it service. I think it makes more sense that space guests aren't a faction, but random, unaffiliated people who arrive by space ship. So instead of faction relationship, there should be a sort of reputation system. Successful visits should improve this reputation, while bad ones worsen it. Only if the reputation is positive should the event happen.
- They can come down by droppod, but RimWorld currently has no system to get back. When selling resources to a space ship, RimWorld does some handwaving and the stuff just disappears. So you could assume that that's also how guests would return to their ship. Some advanced teleportation magi.. I mean, technology.
- This could also be used to avoid crazy exploitation. What if every spacer guest has a "teleportation implant" that will return them safely to the ship, should they be endangered. So if they get attacked, arrested, into a fistfight, etc. it'll just swoop them back up.
- Two kinds of ships:
-- The sort of mass transportation cruise ship kind of guests, that come in droves (think 10, 20, 30 or more guests at once). They have reasonable expectations, but they'll put a load on your resources (they'll still want a place to sleep and if they run out of food they'll take yours).
-- The luxury liner sort. Just a small group, but with very high expectations.
- Consequences:
Obviously, the player doesn't have to worry about faction relationship. But there still have to be consequences. 
-- Very good: A special artifact, as known from missions
-- Good: Rare material maybe?
-- Neutral: You get nothing
-- Bad: They'll pay some bandits to raid you
-- Very bad: Orbital bombardment
- Given the risk/reward nature of the visit, the player should always be asked if he wants to receive spacer guests or not.
- The guests themselves should probably have some kind of fancy pants background. Maybe I can just override their background with "Space traveler".
- Recruitment won't have any relationship costs. So what's the drawback? Or should it be impossible entirely?

**Comment by OrionFive** [comment]
- [x] Spacer guests event base setup
- [x] Spawning pawns
- [ ] Spawning items that guests carry
- [ ] Spacer guest lord
- [ ] Define cruise ship and luxury ship
- [ ] Pawn kinds for cruise guests and luxury guests
- [ ] Arrival dialog
- [ ] Arrive effect
- [ ] Disappear effect
- [ ] Spacer guests have teleportation implant
- [ ] Make guests disappear when endangered and register dissatisfaction
- [ ] Make guests disappear when they leave as a group
- [ ] Extra expectations for luxury guests
- [ ] Consequence: Artifact drop
- [ ] Consequence: Material drop
- [ ] Consequence: Raid
- [ ] Consequence: Bombardment
- [ ] Reputation
- [ ] Spacer guest background story
- [ ] Localization

**Comment by XaraRoy** [comment]
Any Idea what it would take to make this save our ship 2 compatible? Would love to have a space station and have guests shuttle up or come in from space.

**Comment by OrionFive** [comment]
Well as you can see above it's still a long way from complete, and it's also on hold for the time being. 

Also, for something like save our ship 2, it would have to work entirely differently anyway.

**Comment by XaraRoy** [comment]
I asked over there and they said that patches would likely have to be made on this side. I found that strange bc they are not open source. making development on a patch much harder?

**Comment by OrionFive** [comment]
Nah, not much. Usually I decompile everything anyway (RimWorld isn't open source either).But if it's open source, others can actually suggest changes, so...This is not gonna be trivial, I think, so don't hold your horses for this feature. On 10/3/20, 18:04 Xander Roy <notifications@github.com> wrote:

   
  I asked over there and they said that patches would likely have to be made on this side. I found that strange bc they are not open source. making development on a patch much harder?
